tt_metal/host_api.hpp

```diff
@@ -352,37 +351,47 @@ void EnqueueProgram(CommandQueue& cq, std::variant<std::reference_wrapper<Progra
 void Finish(CommandQueue& cq);
 
 /**
- * Creates a trace object which can be used to record commands that have been run. This
- * trace can later be replayed without the further need to create more commands.
- * Return value: trace
+ * Begins capture on a trace, when the trace is in capture mode all programs push into the trace queue will not be executed.
+ * The capture must be later ended via EndTrace, and can be instantiated via InstantiateTrace on a device command queue.
+ * Return value: CommandQueue&
  * | Argument     | Description                                                            | Type                          | Valid Range                        | Required |
  * |--------------|------------------------------------------------------------------------|-------------------------------|------------------------------------|----------|
- * | cq           | The command queue object which dispatches the command to the hardware  | CommandQueue &                |                                    | Yes      |
+ * | trace        | Trace in which to initiate the capture                                 | Trace &                       |                                    | Yes      |
 */
-Trace BeginTrace(CommandQueue& cq);
+CommandQueue& BeginTrace(Trace &trace);
 
 /**
- * This completes a trace and allows it to be replayed. WARNING: Once a trace has been
- * completed for a given command queue, the command queue can no longer be used in eager
- * mode (the default, non tracing mode). This would be undefined behaviour.
+ * Completes capture on a trace, if captured commands do not conform to the rules of the trace, the trace will be invalidated.
+ * This trace can later be instantiated via InstantiateTrace on a device command queue, and executed via EnqueueTrace on the same device command queue.
```

Comment:
this comment says "executed", the comment below says "before the trace can be enqueued".  Seems inconsistent.  Do we enqueue a trace or execute a trace or maybe enqueue to execute a trace?

Reply:
Resolved with 30d3f5eec6316f2b316d7442fe2409a2cd9d5635, now I use consistent phrasing "enqueued for execution" in both places